python/xorbits/deploy/kubernetes/config.py

```diff
@@ -93,10 +93,16 @@ def build(self):
             "metadata": {"name": self._name, "namespace": self._namespace},
             "rules": [
                 {
-                    "apiGroups": self._api_groups,
-                    "resources": self._resources,
-                    "verbs": self._verbs,
-                }
+                    "apiGroups": ["", "apps"],
+                    "resources": [
+                        "pods",
+                        "endpoints",
+                        "services",
+                        "deployments",
+                        "deployments/scale",
+                    ],
+                    "verbs": ["get", "watch", "list", "patch"],
+                },
```

Comment:
Revert to the previous code. The newly added role is passed from the outside or added by default.